Shopify/cli
# Do not proxy `/localization` and `/cart/*` requests

### WHY are these changes introduced?

Fixes #1537
Fixes #1078

When a Theme Access App token (i.e. starting with `shptka_`) is used as `--password`, CLI Proxy sends all requests to `theme-kit-access.shopifyapps.com/cli/sfr`, instead of `<shop name>.myshopify.com`.

However, features behind `/localization` and `/cart/*` are not available in SFR, so that `400` or `404` errors appear.

### WHAT is this pull request doing?

Continues to send `/localization` and `/cart/*` requests to `<shop name>.myshopify.com`.

### How to test your changes?

Make sure to test with shops for which…
1. you have a Theme Access App token
2. at least 1 product that you can add to a cart
3. at least 2 languages, so that you can test the language selector.

To make a `cart/add.js` request, you can use a Dawn theme and…
1. Open `snippets/buy-buttons.liquid` and add `<button id="add-to-cart-via-fetch">Add to cart via fetch</button>` above `{%- form 'product',`
2. Open `global.js` and append:
```JS
const button = document.getElementById('add-to-cart-via-fetch')
if (button) {
  button.addEventListener('click', () => {
    fetch('/cart/add.js', {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({
        items: [
          {
            id: YOUR_PRODUCT_ID,
            quantity: 4,
          },
        ],
      }),
    })
      .then((response) => console.info(response))
      .catch((error) => console.error(error))
  })
}
```

And then continue with…
1. `shopify auth logout`
2. Make sure to be logged off from any admin, or test in an incognito window
3. Open `127.0.0.1:9292`, your browser console’s Network tab, make sure that log is preserved, and then select a new language.
4. Verify that a call to `/localization` has been made and that it responds with `302 Found` and redirects to the selected language

Let's add a product:
1. Open `127.0.0.1:9292`, with your browser console open
2. Navigate to a random product and add it to your cart via your custom button.
3. Verify that the XHR succeeds.

### Measuring impact

How do we know this change was effective? Please choose one:


### Checklist
